src/scripts/import-sdr-shipments-sheet/add-countries.ts

```diff
@@ -0,0 +1,260 @@
+  Object.keys(resultsMap).forEach((key) => {
+    console.log(`    ${key}: ${resultsMap[key].length}`);
+
+    // NOTE: uncomment & set the status key to debug different types of results
+    // if (key !== UploadWorkflowStatus.SUCCESS && key !== UploadWorkflowStatus.ALREADY_EXISTS) {
+    //   resultsMap[key].forEach((result) => {
+    //     console.log(result)
+    //     console.log("\n")
+    //   })
+    // }
+  });
+
+  console.log("Adding items completed!");
+
+  const validCountries: Country[] = [
+    ...resultsMap[UploadWorkflowStatus.SUCCESS],
+    ...resultsMap[UploadWorkflowStatus.ALREADY_EXISTS],
+  ].reduce((countries: Country[], workflow: CountryUploadWorkflow) => {
+    return [...countries, workflow.data];
+  }, [] as Country[]);
+
+  return validCountries;
+}
+
+const isFulfilled = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseFulfilledResult<T> => {
+  return value.status === "fulfilled";
+};
+
+const _isRejected = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseRejectedResult => {
+  return value.status === "rejected";
+};
+
+/* Consolidate Countries
+ * ------------------------------------------------------ */
+function consolidateCountries(shipments: ShipmentCsv[]) {
+  const countries = shipments.reduce((countries: Set<string>, shipment) => {
+    if (shipment.sendingCountry !== "") {
+      countries.add(shipment.sendingCountry);
+    }
+
+    if (shipment.receivingCountry !== "") {
+      countries.add(shipment.receivingCountry);
+    }
+
+    return countries;
+  }, new Set<string>());
+
+  return countries;
+}
+
+/* Parse Country
+ * ------------------------------------------------------ */
+function parseCountry({
+  data,
+  orig,
+  status,
+  logs,
+}: CountryUploadWorkflow): CountryUploadWorkflow {
+  logs = [...logs, `Log: parsing Geo.Country "${orig}"`];
+
+  const code = orig.toUpperCase();
+  const name = lookup.byIso(code)?.country;
+  data = {
+    ...data,
+    code,
+    name,
+  };
+
+  if (name === undefined) {
+    throw {
+      data,
+      orig,
+      status: UploadWorkflowStatus.ORIGINAL_DATA_INVALID,
+      logs: [
+        ...logs,
+        `Error: "${code}" is an invalid ISO 3166 3-digit country code.`,
+      ],
+    };
+  }
+
+  return {
+    data,
+    orig,
+    status,
+    logs,
+  };
+}
+
+/* Get Country
+ * ------------------------------------------------------ */
+async function getCountry({
+  data,
+  orig,
+  status,
+  logs,
+}: CountryUploadWorkflow): Promise<CountryUploadWorkflow> {
+  logs = [...logs, `Log: Checking if Geo.Country "${orig}" already exists.`];
+
+  const query = qs.stringify({
+    filters: {
+      code: {
+        $eq: data.code,
+      },
+    },
+  });
+
+  const response = await fetch(`${STRAPI_ENV.URL}/countries?${query}`, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${STRAPI_ENV.KEY}`,
+    },
+  });
+  const body = await response.json();
```

Comment:
Might want to wrap this in a try/catch so you can add more information to the error if we can't parse the response body.